modules/core/src/test/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotConsensusFunctionsSuite.scala

```diff
@@ -110,44 +138,59 @@ object GlobalSnapshotConsensusFunctionsSuite extends MutableIOSuite with Checker
-      Signed.forAsyncKryo[IO, GlobalSnapshot](genesis, keyPair).flatMap { signedGenesis =>
-        GlobalIncrementalSnapshot.fromGlobalSnapshot(signedGenesis.value).flatMap { lastArtifact =>
-          Signed.forAsyncKryo[IO, GlobalIncrementalSnapshot](lastArtifact, keyPair).flatMap { signedLastArtifact =>
-            mkStateChannelEvent().flatMap { scEvent =>
-              gscf
-                .createProposalArtifact(
-                  SnapshotOrdinal.MinValue,
-                  signedLastArtifact,
-                  signedGenesis.value.info,
-                  EventTrigger,
-                  Set(scEvent.asLeft[DAGEvent]),
-                  facilitators
-                )
-                .flatMap {
-                  case (artifact, _, _) =>
-                    gscf.validateArtifact(signedLastArtifact, signedGenesis.value.info, EventTrigger, artifact, facilitators).map {
-                      result =>
-                        expect.same(result.isRight, true) && expect
-                          .same(
-                            result.map(_._1.stateChannelSnapshots(scEvent.address)),
-                            Right(NonEmptyList.one(scEvent.snapshotBinary))
-                          )
-                    }
-                }
+    Ref.of(List.empty[ForkInfo]).map(mkMockGossip).flatMap { r =>
+      val gscf = mkGlobalSnapshotConsensusFunctions(r)
+
+      val facilitators = Set.empty[PeerId]
+
+      KeyPairGenerator.makeKeyPair[IO].flatMap { keyPair =>
+        val genesis = GlobalSnapshot.mkGenesis(Map.empty, EpochProgress.MinValue)
+        Signed.forAsyncKryo[IO, GlobalSnapshot](genesis, keyPair).flatMap { signedGenesis =>
+          GlobalIncrementalSnapshot.fromGlobalSnapshot(signedGenesis.value).flatMap { lastArtifact =>
+            Signed.forAsyncKryo[IO, GlobalIncrementalSnapshot](lastArtifact, keyPair).flatMap { signedLastArtifact =>
+              mkStateChannelEvent().flatMap { scEvent =>
+                gscf
+                  .createProposalArtifact(
+                    SnapshotOrdinal.MinValue,
+                    signedLastArtifact,
+                    signedGenesis.value.info,
+                    EventTrigger,
+                    Set(scEvent.asLeft[DAGEvent]),
+                    facilitators
+                  )
+                  .flatMap {
+                    case (artifact, _, _) =>
+                      gscf.validateArtifact(signedLastArtifact, signedGenesis.value.info, EventTrigger, artifact, facilitators).map {
+                        result =>
+                          expect.same(result.isRight, true) && expect
+                            .same(
+                              result.map(_._1.stateChannelSnapshots(scEvent.address)),
+                              Right(NonEmptyList.one(scEvent.snapshotBinary))
+                            )
+                      }
+                  }
+              }
             }
```

Comment:
Since we have a lot of nested `flatMap`s then we should consider for-comp:

```scala
for {
  keyPair <- KeyPairGenerator.makeKeyPair[IO]
  genesis = GlobalSnapshot.mkGenesis(Map.empty, EpochProgress.MinValue)
  signedGenesis <- Signed.forAsyncKryo[IO, GlobalSnapshot](genesis, keyPair)
  ... // etc etc
} yield ()
```

Reply:
Re-factored to use `for`-comprehension.